google/cloud/bigquery/job/base.py

```diff
@@ -88,6 +89,22 @@ def _error_result_to_exception(error_result):
 )
 
 
+class TransactionInfo(typing.NamedTuple):
```

Comment:
Does `NamedTuple` make sense here? I suspect we don't want `TransactionInfo` to be a sequence / iterable in our public interface.

Perhaps a data class would be more appropriate. We might have to use the backport for Python 3.6 support. https://pypi.org/project/dataclasses/

Reply:
Or we can just use our usual pattern of a private `_properties` and public properties that pull out the relevant fields.

Reply:
I think a (named) tuple very much makes sense here, yes, since it's a read-only property set on the server. A (frozen) dataclass could do the trick as well, but then it's not that much different from a namedtuple. Ditto for a traditional class with `_properties`, same thing but with more code.

Do you still believe we should change the implementation here?

Reply:
As discussed offline - we'll go with the namedtuple.